From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: julian-siebert <mail@julian-siebert.de>
Date: Thu, 11 Jul 2024 12:55:56 +0200
Subject: [PATCH] Rebrand


diff --git a/src/main/java/io/papermc/paper/ServerBuildInfo.java b/src/main/java/io/papermc/paper/ServerBuildInfo.java
index 7107eabf02d9d9b69d97c978ba421ad52031ec3f..9fbd1e967e4e51b8ab036b5f8ca71e8088fa19ef 100644
--- a/src/main/java/io/papermc/paper/ServerBuildInfo.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfo.java
@@ -25,6 +25,13 @@ public interface ServerBuildInfo {
     Key BRAND_PURPUR_ID = Key.key("purpurmc", "purpur");
     // Purpur end
 
+    // Curepur start
+    /**
+     * The brand id for Curepur.
+     */
+    Key BRAND_CUREPUR_ID = Key.key("uroria", "curepur");
+    // Curepur end
+
     /**
      * Gets the {@code ServerBuildInfo}.
      *
