From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: julian-siebert <mail@julian-siebert.de>
Date: Sat, 27 Jul 2024 08:58:08 +0000
Subject: [PATCH] Publish-Maven-Central


diff --git a/build.gradle.kts b/build.gradle.kts
index ea3ad97909de951efa0ad322df766689339b8af6..6e7ccf7c303ede2225f85f2052b8d2346b5102c4 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -1,7 +1,13 @@
+import net.thebugmc.gradle.sonatypepublisher.PublishingType.* // Curepur
+
 plugins {
     `java-library`
     `maven-publish`
     idea // Paper
+    // Curepur start
+    signing
+    id("net.thebugmc.gradle.sonatype-central-portal-publisher") version("1.2.3")
+    // Curepur end
 }
 
 java {
@@ -100,7 +106,7 @@ configurations {
     for (config in outgoing) {
         config {
             outgoing {
-                capability("${project.group}:${project.name}:${project.version}")
+                capability("${rootProject.group}:${project.name}:${project.version}")
                 capability("io.papermc.paper:paper-mojangapi:${project.version}")
                 capability("com.destroystokyo.paper:paper-mojangapi:${project.version}")
             }
@@ -132,7 +138,7 @@ val generateApiVersioningFile by tasks.registering {
 
 tasks.jar {
     from(generateApiVersioningFile.map { it.outputs.files.singleFile }) {
-        into("META-INF/maven/${project.group}/${project.name}")
+        into("META-INF/maven/${rootProject.group}/${project.name}")
     }
     manifest {
         attributes(
@@ -223,3 +229,60 @@ tasks.check {
     dependsOn(scanJarForOldGeneratedCode)
 }
 // Paper end
+
+/// Curepur start
+val channel = System.getenv("PROJECT_CHANNEL") ?: "dev" // dev (local), stable, latest
+
+centralPortal {
+    username = System.getenv("SONATYPE_USERNAME")
+    password = System.getenv("SONATYPE_PASSWORD")
+
+    publishingType = if (channel == "latest") AUTOMATIC else USER_MANAGED
+
+    name = "curepur-api"
+
+    pom {
+        name = "curepur-api"
+        url = "https://github.com/uroria/curepur"
+        description = "Curepur API"
+
+        licenses {
+            license {
+                name = "Apache 2.0"
+                url = "https://github.com/uroria/curepur/blob/main/LICENSE"
+            }
+        }
+
+        developers {
+            developer {
+                id = "julian-siebert"
+                name = "Julian Siebert"
+                organization = "Uroria"
+                organizationUrl = "https://github.com/uroria"
+                email = "mail@julian-siebert.de"
+            }
+        }
+        scm {
+            connection = "scm:git:git://github.com/uroria/curepur.git"
+            developerConnection = "scm:git:git@github.com:uroria/curepur.git"
+            url = "https://github.com/uroria/curepur"
+            tag = "HEAD"
+        }
+
+        ciManagement {
+            system = "GitHub Actions"
+            url = "https://github.com/uroria/curepur/actions"
+        }
+    }
+}
+
+signing {
+    isRequired = System.getenv("CI") != null
+
+    val privateKey = System.getenv("GPG_PRIVATE_KEY")
+    val passphrase = System.getenv("GPG_PASSPHRASE")
+    useInMemoryPgpKeys(privateKey, passphrase)
+
+    sign(publishing.publications)
+}
+/// Curepur end
\ No newline at end of file
