From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: julian-siebert <mail@julian-siebert.de>
Date: Sat, 13 Jul 2024 22:19:42 +0200
Subject: [PATCH] Root-Module-SHADE


diff --git a/build.gradle.kts b/build.gradle.kts
index 1f5963a0c2868f84906ccea3b641b034a8b3bd9d..87a0edc61215a816a9e83e80e7dba57045d4b013 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -13,6 +13,10 @@ configurations.named(log4jPlugins.compileClasspathConfigurationName) {
 val alsoShade: Configuration by configurations.creating
 
 dependencies {
+    // Curepur start
+    implementation(project(":"))
+    alsoShade(project(":"))
+    // Curepur end
     implementation(project(":curepur-api")) // Purpur // Curepur
     // Paper start
     implementation("org.jline:jline-terminal-jansi:3.21.0")
@@ -136,6 +140,7 @@ tasks.serverJar {
             }
         }
     })
+    duplicatesStrategy = DuplicatesStrategy.EXCLUDE // Curepur - Required Root module MANIFEST Exclusion
 }
 // Paper end - use TCA for console improvements
 
